internal/stores/hostdb.go

```diff
@@ -313,6 +313,40 @@ func (ss *SQLStore) Host(hostKey consensus.PublicKey) (hostdb.Host, error) {
 	return h.convert(), tx.Error
 }
 
+// HostsForScanning returns the address of hosts for scanning.
+func (ss *SQLStore) HostsForScanning(maxLastScan time.Time, offset, limit int) ([]hostdb.HostAddress, error) {
+	if offset < 0 {
+		return nil, ErrNegativeOffset
+	}
+
+	var hosts []struct {
+		PublicKey  consensus.PublicKey `gorm:"unique;index;type:bytes;serializer:gob;NOT NULL"`
+		NetAddress string
+	}
+	var hostAddresses []hostdb.HostAddress
+
+	err := ss.db.
+		Scopes(ExcludeBlockedHosts).
+		Offset(offset).
```

Comment:
you can't apply the offset and limit before the where condition I don't think? unless `gorm` is clever about it and fixes this for you, but regardless it should be between `where` and `FindInBatches`

Reply:
Gorm is smart about building queries but I moved it.